extensions/common/iam/identity-trust/identity-trust-sts/identity-trust-sts-core/src/test/java/org/eclipse/edc/iam/identitytrust/sts/defaults/StsClientTokenIssuanceIntegrationTest.java

```diff
@@ -70,8 +72,11 @@ void setup() {
         privateKeyResolver = new VaultPrivateKeyResolver(keyParserRegistry, vault, mock(), mock());
 
         tokenGeneratorService = new StsClientTokenGeneratorServiceImpl(
-                client -> new JwtGenerationService(),
-                stsClient -> privateKeyResolver.resolvePrivateKey(stsClient.getPrivateKeyAlias()).orElse(null),
+                client -> new JwtGenerationService(keyId -> {
+                    var pk = privateKeyResolver.resolvePrivateKey(keyId).orElse(null);
+                    return CryptoConverter.createSignerFor(pk);
```

Comment:
this is called only here: it could be inlined here, as there's no remaining logic in this provider implementation